test/unit/gemm/device/gemm_testbed_3x.hpp

```diff
@@ -1413,12 +1419,21 @@ struct TestbedImpl {
       }
     }
 
+#if defined(CUTLASS_ENABLE_SYCL)
+    try {
+      syclcompat::wait_and_throw();
+    } catch (std::exception const &e) {
+      ADD_FAILURE() << "Error at Kernel Sync.";
+      return false;
+    }
```

Comment:
Slightly verbose approach here. @FMarno suggested `EXPECT_NO_THROW` , which would be a lot neater, but to match the upstream logic we need to `return false`. :shrug:

Reply:
Why did you not decide to catch exceptions in the sycl namespace explicitly?

You could also use `...` for a catch-all.

Reply:
Hmm, well this will catch all `sycl::exception`s as well, since that derives from `std::exception`. Do you think I ought to treat `std::exception`s differently?

Reply:
no, not really. I just remember seeing it in your first implementation. This is fine.